### ISSUE TITLE
goog.events.dispatchEvent is not a 1:1 replacement for jQuery's `trigger()`

### ISSUE THREAD

**Issue by thedavidmeister** [hoplon/hoplon#188]
I'm getting errors with the simple case `(input :select true)` when using hoplon.goog's implementation of `do!`:

```
(defmethod do! :select
  [elem _ _]
  (events/dispatchEvent elem goog.events.EventType.SELECT))
```

The error I'm seeing is this:

`Assertion failed: Can not use goog.events.dispatchEvent with non-goog.events.Listenable instance.`

From my testing/googling, and this SO post https://stackoverflow.com/questions/23948201/dispatch-event-on-the-angular-element-in-jasmine-test-is-not-working it appears that google closure doesn't actually provide a wrapper around the DOM events like jQuery does, but a totally separate implementation of "events".

The error is basically saying that you can't dispatch events on DOM elements because Google Closure's event dispatching system isn't implemented for DOM elements.

**Comment by flyboarder** [comment]
We can fix this by replacing the `on!` implementation for `hoplon.goog`.

**Comment by flyboarder** [comment]
`on!` has been replaced for a while, we need to duplicate tests between hoplon.jquery and hoplon.goog.

**Comment by thedavidmeister** [comment]
@flyboarder yes, well ideally not duplicate - it would be great to run the same tests twice and "inject" the relevant namespaces at the start of the suite run.

**Comment by flyboarder** [comment]
Any ideas how injecting the namespace could be done?

**Comment by thedavidmeister** [comment]
haha nope, just something that sounds like it would be handy ;)

**Comment by thedavidmeister** [comment]
let's kick this forward to 7.3+
